### ISSUE TITLE
Doesn't build under x86_64

### ISSUE THREAD

**Issue by diversys** [HaikuArchives/SuperPrefs#7]
```
~/SuperPrefs> make
mkdir -p objects.x86_64-cc5-release; \
mkdepend  -I./  -Iobjects.x86_64-cc5-release/  -I. -p .cpp:objects.x86_64-cc5-release/%n.o -m -f "objects.x86_64-cc5-release/App.d" App.cpp
mkdir -p objects.x86_64-cc5-release; \
mkdepend  -I./  -Iobjects.x86_64-cc5-release/  -I. -p .cpp:objects.x86_64-cc5-release/%n.o -m -f "objects.x86_64-cc5-release/MainWindow.d" MainWindow.cpp
g++ -c MainWindow.cpp -iquote./  -iquote./  -iquoteobjects.x86_64-cc5-release/  -iquote.  -isystem/boot/system/develop/headers/private/interface  -isystem/boot/system/develop/headers/private/shared  -O0    -o "objects.x86_64-cc5-release/MainWindow.o"
In file included from MainWindow.cpp:9:0:
MainWindow.h:89:23: error: 'vector' has not been declared
       void  FlatFalse(vector<BString>& vTemp);
                       ^
MainWindow.h:89:29: error: expected ',' or '...' before '<' token
       void  FlatFalse(vector<BString>& vTemp);
                             ^
MainWindow.h:134:11: error: 'set' does not name a type
           set<BString>   sAllPreflets;
           ^
MainWindow.h:135:11: error: 'vector' does not name a type
           vector<BString>   vSign;    
           ^
MainWindow.h:136:4: error: 'vector' does not name a type
    vector<BString>   vPath;    
    ^
MainWindow.h:137:4: error: 'vector' does not name a type
    vector<BString>   vName;
    ^
MainWindow.h:138:4: error: 'vector' does not name a type
    vector<BString>   vSystemApps;
    ^
MainWindow.h:139:4: error: 'vector' does not name a type
    vector<BString>   vAppsSign;    
    ^
MainWindow.h:140:4: error: 'vector' does not name a type
    vector<BString>   vAppsPath;    
    ^
MainWindow.h:141:4: error: 'vector' does not name a type
    vector<BString>   vAppsName;
    ^
MainWindow.h:142:4: error: 'vector' does not name a type
    vector< pair<BString, BString> > RelatedKeywords;
    ^
MainWindow.h:143:4: error: 'vector' does not name a type
    vector<BString>   vTemp;   
    ^
MainWindow.h:144:4: error: 'map' does not name a type
    map<string, string>  PrefsKeyword;   
    ^
MainWindow.h:145:4: error: 'map' does not name a type
    map<BString, BString> NameSign; 
    ^
MainWindow.h:146:4: error: 'map' does not name a type
    map<BString, BString> AppsNameSign;
    ^
MainWindow.h:147:4: error: 'map' does not name a type
    map<BString, BButton*> NameButton;
    ^
MainWindow.h:148:4: error: 'map' does not name a type
    map<BString, BButton*> NameButtonAlphabetical;
    ^
MainWindow.h:149:4: error: 'map' does not name a type
    map<BString, BButton*> NameButtonApps;
    ^
MainWindow.cpp: In member function 'void MainWindow::Keywords()':
MainWindow.cpp:33:2: error: 'RelatedKeywords' was not declared in this scope
  RelatedKeywords.push_back(make_pair(BString("application/x-vnd.Haiku-TrackerPreferences"),BString("File Explorer Settings")));
  ^
MainWindow.cpp:33:125: error: 'make_pair' was not declared in this scope
  RelatedKeywords.push_back(make_pair(BString("application/x-vnd.Haiku-TrackerPreferences"),BString("File Explorer Settings")));
                                                                                                                             ^
MainWindow.cpp:33:125: note: suggested alternative:
In file included from /boot/system/develop/tools/lib/gcc/x86_64-unknown-haiku/5.4.0/include/c++/bits/stl_algobase.h:64:,
                 from /boot/system/develop/tools/lib/gcc/x86_64-unknown-haiku/5.4.0/include/c++/vector:60,
                 from /boot/system/develop/headers/os/interface/Alert.h:9,
                 from MainWindow.h:13,
                 from MainWindow.cpp:9:
/boot/system/develop/tools/lib/gcc/x86_64-unknown-haiku/5.4.0/include/c++/bits/stl_pair.h:286:5: note:   'std::make_pai'
     make_pair(_T1 __x, _T2 __y)
     ^
MainWindow.cpp: In member function 'void MainWindow::populateLayout()':
MainWindow.cpp:91:3: error: 'NameButton' was not declared in this scope
   NameButton[fAppName] = button;
   ^
MainWindow.cpp:94:3: error: 'vSystemApps' was not declared in this scope
   vSystemApps.push_back(fAppName);
   ^
MainWindow.cpp:105:3: error: 'NameButton' was not declared in this scope
   NameButton[fAppName] = button;
   ^
MainWindow.cpp:108:3: error: 'vSystemApps' was not declared in this scope
   vSystemApps.push_back(fAppName);
   ^
MainWindow.cpp:119:3: error: 'NameButton' was not declared in this scope
   NameButton[fAppName] = button;
   ^
MainWindow.cpp:122:3: error: 'vSystemApps' was not declared in this scope
   vSystemApps.push_back(fAppName);
   ^
MainWindow.cpp:133:3: error: 'NameButton' was not declared in this scope
   NameButton[fAppName] = button;
   ^
MainWindow.cpp:136:3: error: 'vSystemApps' was not declared in this scope
   vSystemApps.push_back(fAppName);
   ^
MainWindow.cpp:147:3: error: 'NameButton' was not declared in this scope
   NameButton[fAppName] = button;
   ^
MainWindow.cpp:150:3: error: 'vSystemApps' was not declared in this scope
   vSystemApps.push_back(fAppName);
   ^
MainWindow.cpp:153:19: error: 'vSign' was not declared in this scope
  for(int i=0; i < vSign.size(); i++) {
                   ^
MainWindow.cpp:156:19: error: 'vSystemApps' was not declared in this scope
   if (!(std::find(vSystemApps.begin(), vSystemApps.end(), fAppName) != vSystemApps.end()))
                   ^
MainWindow.cpp:163:4: error: 'NameButton' was not declared in this scope
    NameButton[fAppName] = button;
    ^
MainWindow.cpp:172:6: error: 'map' was not declared in this scope
  for(map<BString, BString>::const_iterator it=NameSign.begin(); it != NameSign.end(); ++ it, ++ splitCount) {
      ^
MainWindow.cpp:172:6: note: suggested alternative:
In file included from /boot/system/develop/tools/lib/gcc/x86_64-unknown-haiku/5.4.0/include/c++/map:61:0,
                 from MainWindow.h:18,
                 from MainWindow.cpp:9:
/boot/system/develop/tools/lib/gcc/x86_64-unknown-haiku/5.4.0/include/c++/bits/stl_map.h:96:11: note:   'std::map'
     class map
           ^
MainWindow.cpp:172:17: error: expected primary-expression before ',' token
  for(map<BString, BString>::const_iterator it=NameSign.begin(); it != NameSign.end(); ++ it, ++ splitCount) {
                 ^
MainWindow.cpp:172:26: error: expected primary-expression before '>' token
  for(map<BString, BString>::const_iterator it=NameSign.begin(); it != NameSign.end(); ++ it, ++ splitCount) {
                          ^
MainWindow.cpp:172:27: error: '::const_iterator' has not been declared
  for(map<BString, BString>::const_iterator it=NameSign.begin(); it != NameSign.end(); ++ it, ++ splitCount) {
                           ^
MainWindow.cpp:172:65: error: 'it' was not declared in this scope
  for(map<BString, BString>::const_iterator it=NameSign.begin(); it != NameSign.end(); ++ it, ++ splitCount) {
                                                                 ^
MainWindow.cpp:172:71: error: 'NameSign' was not declared in this scope
  for(map<BString, BString>::const_iterator it=NameSign.begin(); it != NameSign.end(); ++ it, ++ splitCount) {
                                                                       ^
MainWindow.cpp:180:3: error: 'NameButtonAlphabetical' was not declared in this scope
   NameButtonAlphabetical[it->first] = button;
   ^
MainWindow.cpp:196:17: error: expected primary-expression before ',' token
  for(map<BString, BString>::const_iterator it=AppsNameSign.begin(); it != AppsNameSign.end(); ++ it, ++ splitCount) {
                 ^
MainWindow.cpp:196:26: error: expected primary-expression before '>' token
  for(map<BString, BString>::const_iterator it=AppsNameSign.begin(); it != AppsNameSign.end(); ++ it, ++ splitCount) {
                          ^
MainWindow.cpp:196:27: error: '::const_iterator' has not been declared
  for(map<BString, BString>::const_iterator it=AppsNameSign.begin(); it != AppsNameSign.end(); ++ it, ++ splitCount) {
                           ^
MainWindow.cpp:196:69: error: 'it' was not declared in this scope
  for(map<BString, BString>::const_iterator it=AppsNameSign.begin(); it != AppsNameSign.end(); ++ it, ++ splitCount) {
                                                                     ^
MainWindow.cpp:196:75: error: 'AppsNameSign' was not declared in this scope
  for(map<BString, BString>::const_iterator it=AppsNameSign.begin(); it != AppsNameSign.end(); ++ it, ++ splitCount) {
                                                                           ^
MainWindow.cpp:204:3: error: 'NameButtonApps' was not declared in this scope
   NameButtonApps[it->first] = button;
   ^
MainWindow.cpp: In member function 'void MainWindow::fetchPreflets(directory_which)':
MainWindow.cpp:248:3: error: 'vPath' was not declared in this scope
   vPath.push_back(path.Path()); //Pushing path  
   ^
MainWindow.cpp:249:17: error: 'vSign' was not declared in this scope
   if (std::find(vSign.begin(), vSign.end(), sign) == vSign.end())
                 ^
MainWindow.cpp:251:3: error: 'sAllPreflets' was not declared in this scope
   sAllPreflets.insert(sign);
   ^
MainWindow.cpp:253:3: error: 'vName' was not declared in this scope
   vName.push_back(fAppName);  //Pushing name
   ^
MainWindow.cpp:254:3: error: 'NameSign' was not declared in this scope
   NameSign[fAppName]=sign;
   ^
MainWindow.cpp: In member function 'void MainWindow::fetchApps(directory_which)':
MainWindow.cpp:274:3: error: 'vAppsPath' was not declared in this scope
   vAppsPath.push_back(path.Path()); //Pushing path
   ^
MainWindow.cpp:275:3: error: 'vAppsSign' was not declared in this scope
   vAppsSign.push_back(sign);   //Pushing sign
   ^
MainWindow.cpp:277:3: error: 'vAppsName' was not declared in this scope
   vAppsName.push_back(fAppName);  //Pushing name
   ^
MainWindow.cpp:278:3: error: 'AppsNameSign' was not declared in this scope
   AppsNameSign[fAppName]=sign;
   ^
MainWindow.cpp: In member function 'void MainWindow::fSearch()':
MainWindow.cpp:453:2: error: 'vTemp' was not declared in this scope
  vTemp.clear();
  ^
MainWindow.cpp:454:22: error: 'vName' was not declared in this scope
  for(int i = 0 ; i < vName.size() ; i++) 
                      ^
MainWindow.cpp:459:16: error: 'RelatedKeywords' was not declared in this scope
  for(int i=0;i<RelatedKeywords.size();i++) {
                ^
MainWindow.cpp:472:2: error: 'vTemp' was not declared in this scope
  vTemp.clear();
  ^
MainWindow.cpp:473:22: error: 'vAppsName' was not declared in this scope
  for(int i = 0 ; i < vAppsName.size() ; i++) 
                      ^
MainWindow.cpp: At global scope:
MainWindow.cpp:483:23: error: variable or field 'FlatFalse' declared void
 MainWindow::FlatFalse(vector<BString>& vTemp) {
                       ^
MainWindow.cpp:483:23: error: 'vector' was not declared in this scope
MainWindow.cpp:483:23: note: suggested alternative:
In file included from /boot/system/develop/tools/lib/gcc/x86_64-unknown-haiku/5.4.0/include/c++/vector:64:0,
                 from /boot/system/develop/headers/os/interface/Alert.h:9,
                 from MainWindow.h:13,
                 from MainWindow.cpp:9:
/boot/system/develop/tools/lib/gcc/x86_64-unknown-haiku/5.4.0/include/c++/bits/stl_vector.h:214:11: note:   'std::vecto'
     class vector : protected _Vector_base<_Tp, _Alloc>
           ^
MainWindow.cpp:483:37: error: expected primary-expression before '>' token
 MainWindow::FlatFalse(vector<BString>& vTemp) {
                                     ^
MainWindow.cpp:483:40: error: 'vTemp' was not declared in this scope
 MainWindow::FlatFalse(vector<BString>& vTemp) {
                                        ^
/boot/system/develop/etc/makefile-engine:291: recipe for target 'objects.x86_64-cc5-release/MainWindow.o' failed
make: *** [objects.x86_64-cc5-release/MainWindow.o] Error 1
~/SuperPrefs> 

```

**Comment by anirudhmurali** [comment]
Added namespace, please try building it now.

**Comment by diversys** [comment]
Now I get this:

```
~/SuperPrefs> make
g++ -c MainWindow.cpp -iquote./  -iquote./  -iquoteobjects.x86_64-cc5-release/  -iquote.  -isystem/boot/system/develop/headers/private/interface  -isystem/boot/system/develop/headers/private/shared  -O0    -o "objects.x86_64-cc5-release/MainWindow.o"
In file included from MainWindow.cpp:9:0:
MainWindow.h:89:23: error: 'vector' has not been declared
       void  FlatFalse(vector<BString>& vTemp);
                       ^
MainWindow.h:89:29: error: expected ',' or '...' before '<' token
       void  FlatFalse(vector<BString>& vTemp);
                             ^
MainWindow.h:134:4: error: 'set' does not name a type
    set<BString>   sAllPreflets;
    ^
MainWindow.h:135:4: error: 'vector' does not name a type
    vector<BString>   vSign;    
    ^
MainWindow.h:136:4: error: 'vector' does not name a type
    vector<BString>   vPath;    
    ^
MainWindow.h:137:4: error: 'vector' does not name a type
    vector<BString>   vName;
    ^
MainWindow.h:138:4: error: 'vector' does not name a type
    vector<BString>   vSystemApps;
    ^
MainWindow.h:139:4: error: 'vector' does not name a type
    vector<BString>   vAppsSign;    
    ^
MainWindow.h:140:4: error: 'vector' does not name a type
    vector<BString>   vAppsPath;    
    ^
MainWindow.h:141:4: error: 'vector' does not name a type
    vector<BString>   vAppsName;
    ^
MainWindow.h:142:4: error: 'vector' does not name a type
    vector< pair<BString, BString> > RelatedKeywords;
    ^
MainWindow.h:143:4: error: 'vector' does not name a type
    vector<BString>   vTemp;   
    ^
MainWindow.h:144:4: error: 'map' does not name a type
    map<string, string>  PrefsKeyword;   
    ^
MainWindow.h:145:4: error: 'map' does not name a type
    map<BString, BString>  NameSign; 
    ^
MainWindow.h:146:4: error: 'map' does not name a type
    map<BString, BString>  AppsNameSign;
    ^
MainWindow.h:147:4: error: 'map' does not name a type
    map<BString, BButton*>  NameButton;
    ^
MainWindow.h:148:4: error: 'map' does not name a type
    map<BString, BButton*>  NameButtonAlphabetical;
    ^
MainWindow.h:149:4: error: 'map' does not name a type
    map<BString, BButton*>  NameButtonApps;
    ^
MainWindow.cpp: In member function 'void MainWindow::Keywords()':
MainWindow.cpp:33:2: error: 'RelatedKeywords' was not declared in this scope
  RelatedKeywords.push_back(make_pair(BString("application/x-vnd.Haiku-TrackerPreferences"),BString("File Explorer Settings")));
  ^
MainWindow.cpp: In member function 'void MainWindow::populateLayout()':
MainWindow.cpp:91:3: error: 'NameButton' was not declared in this scope
   NameButton[fAppName] = button;
   ^
MainWindow.cpp:94:3: error: 'vSystemApps' was not declared in this scope
   vSystemApps.push_back(fAppName);
   ^
MainWindow.cpp:105:3: error: 'NameButton' was not declared in this scope
   NameButton[fAppName] = button;
   ^
MainWindow.cpp:108:3: error: 'vSystemApps' was not declared in this scope
   vSystemApps.push_back(fAppName);
   ^
MainWindow.cpp:119:3: error: 'NameButton' was not declared in this scope
   NameButton[fAppName] = button;
   ^
MainWindow.cpp:122:3: error: 'vSystemApps' was not declared in this scope
   vSystemApps.push_back(fAppName);
   ^
MainWindow.cpp:133:3: error: 'NameButton' was not declared in this scope
   NameButton[fAppName] = button;
   ^
MainWindow.cpp:136:3: error: 'vSystemApps' was not declared in this scope
   vSystemApps.push_back(fAppName);
   ^
MainWindow.cpp:147:3: error: 'NameButton' was not declared in this scope
   NameButton[fAppName] = button;
   ^
MainWindow.cpp:150:3: error: 'vSystemApps' was not declared in this scope
   vSystemApps.push_back(fAppName);
   ^
MainWindow.cpp:153:19: error: 'vSign' was not declared in this scope
  for(int i=0; i < vSign.size(); i++) {
                   ^
MainWindow.cpp:156:19: error: 'vSystemApps' was not declared in this scope
   if (!(std::find(vSystemApps.begin(), vSystemApps.end(), fAppName) != vSystemApps.end()))
                   ^
MainWindow.cpp:163:4: error: 'NameButton' was not declared in this scope
    NameButton[fAppName] = button;
    ^
MainWindow.cpp:172:47: error: 'NameSign' was not declared in this scope
  for(map<BString, BString>::const_iterator it=NameSign.begin(); it != NameSign.end(); ++ it, ++ splitCount) {
                                               ^
MainWindow.cpp:180:3: error: 'NameButtonAlphabetical' was not declared in this scope
   NameButtonAlphabetical[it->first] = button;
   ^
MainWindow.cpp:196:47: error: 'AppsNameSign' was not declared in this scope
  for(map<BString, BString>::const_iterator it=AppsNameSign.begin(); it != AppsNameSign.end(); ++ it, ++ splitCount) {
                                               ^
MainWindow.cpp:204:3: error: 'NameButtonApps' was not declared in this scope
   NameButtonApps[it->first] = button;
   ^
MainWindow.cpp: In member function 'void MainWindow::fetchPreflets(directory_which)':
MainWindow.cpp:248:3: error: 'vPath' was not declared in this scope
   vPath.push_back(path.Path()); //Pushing path  
   ^
MainWindow.cpp:249:17: error: 'vSign' was not declared in this scope
   if (std::find(vSign.begin(), vSign.end(), sign) == vSign.end())
                 ^
MainWindow.cpp:251:3: error: 'sAllPreflets' was not declared in this scope
   sAllPreflets.insert(sign);
   ^
MainWindow.cpp:253:3: error: 'vName' was not declared in this scope
   vName.push_back(fAppName);  //Pushing name
   ^
MainWindow.cpp:254:3: error: 'NameSign' was not declared in this scope
   NameSign[fAppName]=sign;
   ^
MainWindow.cpp: In member function 'void MainWindow::fetchApps(directory_which)':
MainWindow.cpp:274:3: error: 'vAppsPath' was not declared in this scope
   vAppsPath.push_back(path.Path()); //Pushing path
   ^
MainWindow.cpp:275:3: error: 'vAppsSign' was not declared in this scope
   vAppsSign.push_back(sign);   //Pushing sign
   ^
MainWindow.cpp:277:3: error: 'vAppsName' was not declared in this scope
   vAppsName.push_back(fAppName);  //Pushing name
   ^
MainWindow.cpp:278:3: error: 'AppsNameSign' was not declared in this scope
   AppsNameSign[fAppName]=sign;
   ^
MainWindow.cpp: In member function 'void MainWindow::fSearch()':
MainWindow.cpp:453:2: error: 'vTemp' was not declared in this scope
  vTemp.clear();
  ^
MainWindow.cpp:454:22: error: 'vName' was not declared in this scope
  for(int i = 0 ; i < vName.size() ; i++) 
                      ^
MainWindow.cpp:459:16: error: 'RelatedKeywords' was not declared in this scope
  for(int i=0;i<RelatedKeywords.size();i++) {
                ^
MainWindow.cpp:472:2: error: 'vTemp' was not declared in this scope
  vTemp.clear();
  ^
MainWindow.cpp:473:22: error: 'vAppsName' was not declared in this scope
  for(int i = 0 ; i < vAppsName.size() ; i++) 
                      ^
MainWindow.cpp: At global scope:
MainWindow.cpp:483:1: error: prototype for 'void MainWindow::FlatFalse(std::vector<BString>&)' does not match any in class 'MainWindow'
 MainWindow::FlatFalse(vector<BString>& vTemp) {
 ^
In file included from MainWindow.cpp:9:0:
MainWindow.h:89:13: error: candidate is: void MainWindow::FlatFalse(int)
       void  FlatFalse(vector<BString>& vTemp);
             ^
MainWindow.cpp: In member function 'void MainWindow::FlatTrue()':
MainWindow.cpp:508:22: error: 'vName' was not declared in this scope
  for(int i = 0 ; i < vName.size() ; ++i) {
                      ^
MainWindow.cpp:509:3: error: 'NameButton' was not declared in this scope
   NameButton[vName[i]]->SetFlat(true);
   ^
MainWindow.cpp:512:3: error: 'NameButtonAlphabetical' was not declared in this scope
   NameButtonAlphabetical[vName[i]]->SetFlat(true);
   ^
MainWindow.cpp:518:23: error: 'vAppsName' was not declared in this scope
   for(int i = 0 ; i < vAppsName.size() ; ++ i ) {
                       ^
MainWindow.cpp:519:3: error: 'NameButtonApps' was not declared in this scope
   NameButtonApps[vAppsName[i]]->SetFlat(true);  
   ^
/boot/system/develop/etc/makefile-engine:291: recipe for target 'objects.x86_64-cc5-release/MainWindow.o' failed
make: *** [objects.x86_64-cc5-release/MainWindow.o] Error 1
~/SuperPrefs> 
```

**Comment by richienyhus** [comment]
This can be closed with 5f7ece6 being merged in.
